src/app/shared/repositories/proof/proof.ts

```diff
@@ -41,6 +48,36 @@ export class Proof {
    */
   cameraSource: CameraSource = CameraSource.Camera;
 
+  /**
+   * Used to sort the assets in the VERIFIED tab either by timestamp or uploadedAt (if available).
+   */
+  get uploadedAtOrTimestamp() {
+    const MILLISECONDS_PER_SECOND = 1000;
+    const LENGTH_IN_MILLISECONDS = 13;
+
+    // convert timestamp to milliseconds if needed
+    const proofTimestampInMilliseconds =
+      this.timestamp.toString().length === LENGTH_IN_MILLISECONDS
+        ? this.timestamp
+        : this.timestamp * MILLISECONDS_PER_SECOND;
+
+    try {
+      const serverTimestampInMilliseconds = Date.parse(this.uploadedAt ?? '');
```

Comment:
The try-catch clause seems unnecessary here. `Date.parse` returns `NaN` for empty string or string that cannot be parsed into timestamp.

Reply:
@shc261392, you are right I tested with empty string, invalid date string it return `NaN`. Here is the commit [refactor(proof.ts): remove unused try catch block](https://github.com/numbersprotocol/capture-lite/pull/3158/commits/dbd03540a065537378552dbb24c99e0b4701f96b)